src/libraries/Common/src/Interop/Browser/Interop.Runtime.Api.cs

```diff
@@ -0,0 +1,386 @@
+            if (mb == null)
+                return string.Empty;
+
+            ParameterInfo[] parms = mb.GetParameters();
+            var parmsLength = parms.Length;
+            if (parmsLength == 0)
+                return string.Empty;
+
+            var res = new char[parmsLength];
+
+            for (int c = 0; c < parmsLength; c++)
+            {
+                var t = parms[c].ParameterType;
+                switch (Type.GetTypeCode(t))
+                {
+                    case TypeCode.Byte:
+                    case TypeCode.SByte:
+                    case TypeCode.Int16:
+                    case TypeCode.UInt16:
+                    case TypeCode.Int32:
+                    case TypeCode.UInt32:
+                    case TypeCode.Boolean:
+                        // Enums types have the same code as their underlying numeric types
+                        if (t.IsEnum)
+                            res[c] = 'j';
+                        else
+                            res[c] = 'i';
+                        break;
+                    case TypeCode.Int64:
+                    case TypeCode.UInt64:
+                        // Enums types have the same code as their underlying numeric types
+                        if (t.IsEnum)
+                            res[c] = 'k';
+                        else
+                            res[c] = 'l';
+                        break;
+                    case TypeCode.Single:
+                        res[c] = 'f';
+                        break;
+                    case TypeCode.Double:
+                        res[c] = 'd';
+                        break;
+                    case TypeCode.String:
+                        res[c] = 's';
+                        break;
+                    default:
+                        if (t == typeof(IntPtr))
+                        {
+                            res[c] = 'i';
+                        }
+                        else if (t == typeof(Uri))
+                        {
+                            res[c] = 'u';
+                        }
+                        else
+                        {
+                            if (t.IsValueType)
+                                throw new NotSupportedException("ValueType arguments are not supported.");
+                            res[c] = 'o';
+                        }
+                        break;
+                }
+            }
+            return new string(res);
+        }
+        internal static void SetupJSContinuation(Task task, JSObject continuationObj)
+        {
+            if (task.IsCompleted)
+                Complete();
+            else
+                task.GetAwaiter().OnCompleted(Complete);
+
+            void Complete()
+            {
+                try
+                {
+                    if (task.Exception == null)
+                    {
+                        var resultProperty = task.GetType().GetProperty("Result");
```

Comment:
This should be run only for `if (task.GetType() == typeof(Task<>))` and I think we can cache it's into static PropertyInfo

Reply:
The check above for `if (task.GetType() == typeof(Task<>))` is not necessary as it is validated before getting here in https://github.com/dotnet/runtime/pull/35566 in the `src/mono/wasm/runtime/driver.c` https://github.com/dotnet/runtime/pull/35566/files#diff-ee313276f7a12336872aeeb4dd9ed971R494

Reply:
Will look at caching the PropertyInfo here

Reply:
Addressed in recent commit